### ISSUE TITLE
OrderBook count problems for C#

### ISSUE THREAD

**Issue by skuvv** [ccxt/ccxt#21892]
### CCXT Version

4.2.80

### Description

Two problems with watching OrderBook:
1)Result count mismatch for property ccxt.pro.IOrderBook.bids.Count is always = 0
Result count mismatch for property ccxt.pro.IOrderBook.asks.Count is always = 0
2)When I change the limit2 for a specific symbol, the depth remains the same as when it was first called.

### Code

```
Console.WriteLine("starting");

var exchange = new ccxt.pro.binance();

var task1 = Task.Run(async () =>
{
    try
    {
        var requestCount1 = 5;
        var response1 = await exchange.WatchOrderBook("BTC/USDT", limit2: requestCount1);
        if (response1.TryGetValue("bids", out var bids1) && bids1 is ccxt.pro.Bids _bids1)
        {
            Console.WriteLine($"Response1:");
            Console.WriteLine($"Actual count: {_bids1.Count}");
            Console.WriteLine($"Required count: {requestCount1}");
            Console.WriteLine($"Property count mismatch:{response1.bids.Count != _bids1.Count}");
            Console.WriteLine($"Required count mismatch:{_bids1.Count != requestCount1}");
        }

        var requestCount2 = 7;
        var response2 = await exchange.WatchOrderBook("BTC/USDT", limit2: requestCount2);
        if (response2.TryGetValue("bids", out var bids2) && bids2 is ccxt.pro.Bids _bids2)
        {
            Console.WriteLine($"\r\nResponse2:");
            Console.WriteLine($"Actual count: {_bids2.Count}");
            Console.WriteLine($"Required count: {requestCount2}");
            Console.WriteLine($"Property count mismatch:{response2.bids.Count != _bids2.Count}");
            Console.WriteLine($"Required count mismatch:{_bids2.Count != requestCount2}");
        }

        var requestCount3 = 3;
        var response3 = await exchange.WatchOrderBook("BTC/USDT", limit2: requestCount3);
        if (response3.TryGetValue("bids", out var bids3) && bids3 is ccxt.pro.Bids _bids3)
        {
            Console.WriteLine($"\r\nResponse3:");
            Console.WriteLine($"Actual count: {_bids3.Count}");
            Console.WriteLine($"Required count: {requestCount3}");
            Console.WriteLine($"Property count mismatch:{response3.bids.Count != _bids3.Count}");
            Console.WriteLine($"Required count mismatch:{_bids3.Count != requestCount3}");
        }
    }
    catch (Exception e)
    {
        Console.WriteLine(e);
    }
});

Console.WriteLine("waiting");
Console.ReadLine();

```

Output:
```
starting
waiting
Response1:
Actual count: 5
Required count: 5
Property count mismatch:True
Required count mismatch:False

Response2:
Actual count: 5
Required count: 7
Property count mismatch:True
Required count mismatch:True

Response3:
Actual count: 5
Required count: 3
Property count mismatch:True
Required count mismatch:True

```

**Comment by carlosmiei** [comment]
Hello @skuvv, thanks for reporting it, we will take a look

**Comment by carlosmiei** [comment]
@skuvv The main count issue should be fixed now, can you please update your ccxt version and try again? (the limit issue is a different one and is transversal to all languages/exchanges, we will tackle it in the future).

**Comment by skuvv** [comment]
Property count is fixed, checked on 4.2.85.
Should I close this issue?
